WordPress/pattern-directory
# Theme: Only load iframe assets when pattern thumbnail is in view.

Loading pattern thumbnails is expensive on page load. This PR implements lazy loading of iframe assets only when the pattern is visible in the viewport.

See #255


### Benchmark
Environment: 
- Chrome Lighthouse (Mobile)
- Localhost
- Incognito mode.



 | Vital | Improvement |
| --- | --- |
 |Time to Interact  | 32%  | 
| First Contentful Paint | 0 |  
| Speed Index |  12% |
| Largest Contentful Paint |  3% |
| Total Blocking Time | 34% |

_Based on the average of 3 test runs before & after._



### Screenshots
[Screencast](https://d.pr/i/63mhH3)



### How to test the changes in this Pull Request:

1. Visit `/`
2. As you scroll down, the visible pattern thumbnails should load their assets.
3. Click different categories and expect all the assets for pattern thumbnails to be loaded.
